overlord/fdestate/backend/reseal.go

```diff
@@ -60,7 +60,10 @@ func MockSecbootBuildPCRProtectionProfile(f func(modelParams []*secboot.SealKeyM
 	}
 }
 
-type StateUpdater func(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error
+type FDEState interface {
+	UpdateState(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error
+	GetState(role string, containerRole string) (bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte, err error)
+}
```

Comment:
```suggestion
type FDEStateManager interface {
	Update(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error
	Get(role string, containerRole string) (bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte, err error)
}
```

depending on how useful it would be for mocking in the tests the interface can be split into `*Getter` (eg. doReseal only uses Get(), and `*Updater` and then have one interface combining the two. Although looking at test changes it's probably not worth it.